7_react/react1/week2/src/components/TodoList.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { Component } from "react";
+import "./TodoList.css";
+
+const todos = [
+  {
+    id: 1,
+    description: "Attend family meeting",
+    deadline: "Sun Mar 10 2024",
+    done: false,
+  },
+  {
+    id: 2,
+    description: "Get some groceries",
+    deadline: "Mon Mar 11 2024",
+    done: false,
+  },
+  {
+    id: 3,
+    description: "Walk my dog",
+    deadline: "Tue Mar 12 2024",
+    done: false,
+  },
+];
+
+class TodoList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: todos,
+      timer: 0,
+      description: "",
+      deadline: "",
+    };
+  }
+
+  componentDidMount() {
+    this.timerID = setInterval(() => this.tick(), 1000);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timerID);
+  }
+
+  tick() {
+    this.setState({
+      timer: this.state.timer + 1,
+    });
+  }
+
+  handleInputChange = (e) => {
+    this.setState({
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  addRandomTodo = () => {
+    const { description, deadline } = this.state;
+    const newTodo = {
+      id: Date.now(),
+      description: description,
+      deadline: deadline,
+      done: false,
+    };
```

Comment:
Right now you can add also empty fields, you should check first whether the input are valid before inserting them :)